bigtable/tests/filters_integration_test.cc

```diff
@@ -0,0 +1,682 @@
+void CheckLatest(bigtable::Client& client, bigtable::Table& table,
+                 std::string const& row_key);
+void CheckFamilyRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckColumnRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckColumnRange(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key);
+void CheckTimestampRange(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key);
+void CheckCellsRowLimit(bigtable::Client& client, bigtable::Table& table,
+                        std::string const& row_key_prefix);
+void CheckCellsRowOffset(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key_prefix);
+void CheckCellsRowSample(bigtable::Client& client, bigtable::Table& table,
+                         std::string const& row_key_prefix);
+}  // anonymous namespace
+
+int main(int argc, char* argv[]) try {
+  namespace admin_proto = ::google::bigtable::admin::v2;
+
+  // Make sure the arguments are valid.
+  if (argc != 4) {
+    std::string const cmd = argv[0];
+    auto last_slash = std::string(argv[0]).find_last_of("/");
+    std::cerr << "Usage: " << cmd.substr(last_slash + 1)
+              << " <project> <instance> <table>" << std::endl;
+    return 1;
+  }
+  std::string const project_id = argv[1];
+  std::string const instance_id = argv[2];
+  std::string const table_name = argv[3];
+  std::string const family = "fam";
+  std::string const fam0 = "fam0";
+  std::string const fam1 = "fam1";
+  std::string const fam2 = "fam2";
+  std::string const fam3 = "fam3";
+
+  auto admin_client =
+      bigtable::CreateAdminClient(project_id, bigtable::ClientOptions());
+  bigtable::TableAdmin admin(admin_client, instance_id);
+
+  auto table_list = admin.ListTables(admin_proto::Table::NAME_ONLY);
+  if (not table_list.empty()) {
+    std::ostringstream os;
+    os << "Expected empty instance at the beginning of integration test";
+    throw std::runtime_error(os.str());
+  }
+
+  auto created_table = admin.CreateTable(
+      table_name,
+      bigtable::TableConfig({{family, bigtable::GcRule::MaxNumVersions(10)},
+                             {fam0, bigtable::GcRule::MaxNumVersions(10)},
+                             {fam1, bigtable::GcRule::MaxNumVersions(10)},
+                             {fam2, bigtable::GcRule::MaxNumVersions(10)},
+                             {fam3, bigtable::GcRule::MaxNumVersions(10)}},
+                            {}));
+  std::cout << table_name << " created successfully" << std::endl;
+
+  bigtable::Client client(project_id, instance_id);
+  std::unique_ptr<bigtable::Table> table = client.Open(table_name);
+
+  CheckPassAll(client, *table, "aaa0001-pass-all");
+  // TODO(google-cloud-go#839) - remove workarounds for emulator bug(s).
```

Comment:
Maybe `google-cloud-go/issues/839`? That's the final piece of the full URL:

```
https://github.com/googlecloudplatform/google-cloud-go/issues/839
```

Reply:
The full link eats almost all 80 characters.  Created an issue in google-cloud-cpp (#151) to track this, and pointed to it from the code.  Our issue has links to the issues open with the emulator.